### ISSUE TITLE
Implement parameter sweep and walk-forward validation

### ISSUE THREAD

**Issue by manikolbe** [manikolbe/aurex-trade#24]
## Summary

Build parameter optimization tooling on top of the backtesting framework (#22). Enables grid search over strategy parameters and walk-forward validation to prevent overfitting.

## Depends On

- #22 (Backtesting framework) — must be completed first

### Parameter Sweep (Grid Search)
- Accept a `strategy_factory` callable (e.g., `lambda short, long: SMACrossover(short, long)`)
- Define parameter grids (e.g., short=[5,10,20], long=[20,50,100])
- Run backtest for every combination
- Rank results by Sharpe ratio, total P&L, or other metrics
- Output comparison table

### Walk-Forward Validation
- Split historical data into rolling train/test windows
- On each window: optimize parameters on training set, validate on test set
- Prevents overfitting to historical data
- Reports aggregate out-of-sample performance

### CLI
```
just sweep --strategy sma_crossover --short 5,10,20 --long 20,50,100 --start 2025-01-01 --end 2025-12-31
just walk-forward --strategy sma_crossover --windows 5 --train-ratio 0.7
```

## Design

- `src/aurex_trade/backtest/sweep.py` — ParameterSweep class with strategy_factory pattern
- `src/aurex_trade/backtest/walk_forward.py` — WalkForwardValidator
- Strategy-agnostic: works with any Strategy protocol implementor via factory callable
- Reuses BacktestRunner from #22 for each individual run

## Acceptance Criteria

- [ ] Grid search over arbitrary strategy parameters
- [ ] Results ranked by multiple metrics
- [ ] Walk-forward validation with configurable window count and train ratio
- [ ] Out-of-sample metrics reported separately from in-sample
- [ ] Works with any strategy implementing the Strategy Protocol
- [ ] Deterministic with seed
- [ ] `just check` passes

**Comment by manikolbe** [comment]
Implemented in e65bc77. Parameter sweep (grid search) and walk-forward validation with non-overlapping train/test windows. Verified end-to-end on 3 months of XAU_USD M1 data.
